backend/src/routes/api/modelRegistrySettings/secrets/index.ts

```diff
@@ -0,0 +1,92 @@
+import { FastifyReply, FastifyRequest } from 'fastify';
+import { V1Secret } from '@kubernetes/client-node';
+import { secureAdminRoute } from '../../../../utils/route-security';
+import { KubeFastifyInstance, RecursivePartial } from '../../../../types';
+import { MODEL_REGISTRY_NAMESPACE } from '../../../../utils/constants';
+
+export default async (fastify: KubeFastifyInstance): Promise<void> => {
```

Comment:
This should work like any other admin route -- not sure what I'm missing here, why the question?

Reply:
This is a route at `/api/modelRegistrySettings/secrets` which lets the user create, read, and update any secret name but hard-coded (via shared constant) to the odh-model-registries namespace. It was a compromise between having a wide open `/api/secrets` that would let a RHOAI admin mess with secrets in any namespace (security issue) and tightly coupling the secrets logic in the routes for reading/updating model registries themselves on the backend instead of doing it in the frontend (less flexible, harder to refactor later)

Reply:
What kind of flexibility do you need here? What kind of secrets are you modifying? @mturley

Reply:
The ModelRegistry spec has database credentials which include a `passwordSecret`, which is a reference to a Secret containing that db password. We have a create modal that needs to create a secret, then create a modelregistry referencing that secret, then take the uid of the new modelregistry and update the secret to add an ownerReference to it. Then we have a modal to view the database details including the password, so we need to look up the secret referenced by a modelregistry.

My initial proposal was that we modify our backend route for CRUD on modelregistries to take an object with `{ modelRegistry, password }` and handle the secret logic internally on the backend. But that felt a little restrictive since all the other backend services are pretty directly exposing API objects and the business logic is in the frontend, which I understand is good in case we need/want to pivot from having these backend routes and instead do all this through a passthrough API? @lucferbux mentioned something about it being more community-contribution-friendly this way as well, I can't recall the reasoning there.